apps/console/src/features/connections/components/edit/settings/attribute-management/uri-attributes-settings.tsx

```diff
@@ -64,18 +76,50 @@ export const UriAttributesSettings: FunctionComponent<AdvanceAttributeSettingsPr
         updateRole,
         updateSubject,
         roleError,
+        selectedClaimMappings,
         subjectError,
         isReadOnly,
         isMappingEmpty,
+        isOIDC,
         isSaml,
         [ "data-testid" ]: testId
     } = props;
 
     const { t } = useTranslation();
+    const { UIConfig } = useUIConfig();
+
+    const [ groupAttribute, setGroupAttribute ] = useState<string>("");
+
+    useEffect(() => {
+        if (!selectedClaimMappings || selectedClaimMappings.length === 0) {
+            return;
+        }
+
+        setGroupAttribute(getGroupAttribute());
+    }, [ selectedClaimMappings ]);
+
+    useEffect(() => {
+        // If the initial role uri is not available, then use the group attribute.
+        updateRole(initialRoleUri ? initialRoleUri : groupAttribute);
```

Comment:
We can use null check here.
```suggestion
        updateRole(initialRoleUri ?? groupAttribute);
```